storm-core/src/jvm/org/apache/storm/container/cgroup/CgroupCenter.java

```
@@ -0,0 +1,238 @@
+ * limitations under the License.
+ */
+package org.apache.storm.container.cgroup;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class CgroupCenter implements CgroupOperation {
+
+    public static Logger LOG = LoggerFactory.getLogger(CgroupCenter.class);
+
+    private static CgroupCenter instance;
+
+    private CgroupCenter() {
+
+    }
+
+    /**
+     * Thread unsafe
+     * 
+     * @return
+     */
+    public synchronized static CgroupCenter getInstance() {
+        if (instance == null)
+            instance = new CgroupCenter();
+        return CgroupUtils.enabled() ? instance : null;
+    }
+
+    @Override
+    public List<Hierarchy> getHierarchies() {
+
+        Map<String, Hierarchy> hierarchies = new HashMap<String, Hierarchy>();
+        FileReader reader = null;
+        BufferedReader br = null;
+        try {
+            reader = new FileReader(Constants.MOUNT_STATUS_FILE);
+            br = new BufferedReader(reader);
+            String str = null;
+            while ((str = br.readLine()) != null) {
+                String[] strSplit = str.split(" ");
+                if (!strSplit[2].equals("cgroup"))
+                    continue;
+                String name = strSplit[0];
+                String type = strSplit[3];
+                String dir = strSplit[1];
+                Hierarchy h = hierarchies.get(type);
+                h = new Hierarchy(name, CgroupUtils.analyse(type), dir);
+                hierarchies.put(type, h);
+            }
+            return new ArrayList<Hierarchy>(hierarchies.values());
+        } catch (Exception e) {
+            LOG.error("Get hierarchies error", e);
```

I realize a lot of this code came directly from JStorm, but I really would prefer to not eat exceptions.  If CGroups is supposed to be enabled I would rather have the supervisor crash if there is an issue running instead of disabling it and outputting an error message.